### PR TITLE
fix "turn.recognized" does not include entities with "None" intent in Cross-Trained Recognizer Set

### PR DESCRIPTION
**Pull request by cosmicshuai**
Parity with PR:https://github.com/microsoft/botbuilder-dotnet/pull/4655

In this PR, all entities in recognizer results of Cross-Trained Recognizer Set even with "None" intent, will be included in the final result.

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts [hunk 1]
```diff
@@ -120,10 +121,18 @@ export class CrossTrainedRecognizerSet extends Recognizer {
             return recognizerResults[consensusRecognizedId];
         }
 
+        //find if matched entities found when hits the none intent
+        let mergedEntities = {};
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts:125]: Prefer reduce:
```typescript
const mergedEntities = results.reduce((acc, curr) => lodash.merge(acc, curr.entities), {});
```

**Review by joshgummersall — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts (hunk 1 above)
Comment (cosmicshuai) [reply libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts]: great idea. changed to reduce method

File: libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import { RecognizerResult, Activity, getTopScoringIntent } from 'botbuilder-core';
 import { DialogContext } from 'botbuilder-dialogs';
 import { Recognizer } from './recognizer';
+import * as lodash from 'lodash';
```
[the remark this answers is not in the document]
Comment (cosmicshuai) [reply libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts]: changed to 
```JavaScript
import { merge } from 'lodash';
```

File: libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts [hunk 3]
```diff
@@ -132,10 +133,14 @@ export class CrossTrainedRecognizerSet extends Recognizer {
...
         }
 
+        //find if matched entities found when hits the none intent
+        const mergedEntities = results.reduce((acc, curr) => lodash.merge(acc, curr.entities), {});
+
         // return none
         const recognizerResult: RecognizerResult = {
             text,
-            intents: { None: { score: 1.0 } },
+            intents: { 'None': { score: 1.0 } },
```
[the remark this answers is not in the document]
Comment (cosmicshuai) [reply libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts]: removed quotes

**Review by joshgummersall — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ import { Activity, RecognizerResult, getTopScoringIntent } from 'botbuilder-core
 import { Converter, ConverterFactory, DialogContext } from 'botbuilder-dialogs';
 import { RecognizerListConverter } from '../converters';
 import { Recognizer, RecognizerConfiguration } from './recognizer';
+import { merge } from 'lodash';
```
[the remark this answers is not in the document]
Comment (cosmicshuai) [reply libraries/botbuilder-dialogs-adaptive/src/recognizers/crossTrainedRecognizerSet.ts]: understood. 
now it is `import merge from 'lodash/merge`

File: libraries/botbuilder-dialogs-adaptive/tsconfig.json [hunk 5]
```diff
@@ -8,7 +8,9 @@
     "declarationMap": true,
     "outDir": "./lib",
     "rootDir": "./src",
-    "types" : ["node"]
+    "types" : ["node"],
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive/tsconfig.json:13]: These are required for the Lodash import to work properly?

Comment (cosmicshuai) [reply libraries/botbuilder-dialogs-adaptive/tsconfig.json:13]: right, TypeScript will throw error: can only be default-imported using the 'esModuleInterop' flag. 
After these two flag set to true. The error disappeared.

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**